core/lib/src/config/address.rs

```diff
@@ -0,0 +1,60 @@
+use std::{io, fmt};
+use std::str::FromStr;
+use std::path::PathBuf;
+use std::net::{IpAddr, ToSocketAddrs};
+
+/// An enum corresponding to the address to serve on
```

Comment:
This needs a docstring. How about:

```suggestion
/// The configured address to serve a Rocket application on.
///
/// Addresses can be specified as either IP addresses or hostnames. On Unix
/// platforms, addresses can also be paths to Unix domain sockets specified as
/// `unix:path/to/socket`.
///
/// When a hostname is specified, it is resolved to the _first available_ IP
/// address for the given hostname. Rocket _does not_ bind to all addresses
/// available for a given hostname. For example, if both `127.0.0.1` and `::1`
/// are specified for `localhost`, the first address available of the two will
/// be used, and the other address will be ignored by Rocket.
///
/// When a Unix domain socket is specified, a lock file named
/// `path/to/socket.lock` is created and locked. If locking fails, application
/// launch is aborted. Additionally, if `path/to/socket` does not exist, it is
/// created. Both the lock file and the socket file are unconditionally deleted
/// upon server termination.
```